src/Concerns/WaitsForElements.php

```diff
@@ -343,6 +344,38 @@ public function clickAndWaitForReload($selector = null)
         });
     }
 
+    /**
+     * Wait for the given event type to occur on a target.
+     *
+     * @param  string  $type
+     * @param  string|null  $target
+     * @param  int|null  $seconds
+     * @return $this
+     *
+     * @throws \Facebook\WebDriver\Exception\TimeOutException
+     */
+    public function waitForEvent($type, $target = null, $seconds = null)
+    {
+        $seconds = is_null($seconds) ? static::$waitSeconds : $seconds;
+
+        if ($target !== 'document' && $target !== 'window') {
+            $target = $this->resolver->findOrFail($target ?? '');
+        }
+
+        $this->driver->manage()->timeouts()->setScriptTimeout($seconds);
+
+        try {
+            $this->driver->executeAsyncScript(
+                'eval(arguments[0]).addEventListener(arguments[1], () => arguments[2](), { once: true });',
+                [$target, $type]
+            );
+        } catch (ScriptTimeoutException $e) {
+            throw new TimeoutException("Waited {$seconds} seconds for '${type}' event.");
```

Comment:
${type} doesn't look like valid PHP. Did you try this on a real application?

Reply:
That is indeed a typo, sorry about that. I'm using this code in my Laravel/Inertia app, but haven't hit the exception since I changed the message.

I fixed the syntax and made sure to test the exception.